### PR TITLE
[UPDATE] Updates WebRTC and Socket.io client to the latest versions

### PR DESCRIPTION
**Pull request by andreahaku**
- updates `WebRTC` to v1.106.1
- updates `Socket.io` client to 4.5.3

Note: `WebRTC v1.106.1` no longer requires to download the bitcode for iOS so this step has been removed from `package.json` and `bitrise.yml`

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require_relative '../node_modules/react-native/scripts/react_native_pods'
 require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
 
-platform :ios, "11.0"
+platform :ios, "12.0"
```
Comment (Cal-L) [inline ios/Podfile:4]: Is it safe to set the deployment target here to 12 while Xcode's is set to 11? Have you experienced any warnings as a result of the versions differing?

File: bitrise.yml [hunk 2]
```diff
@@ -48,11 +43,6 @@ workflows:
             - cache_local_deps: 'yes'
             - command: setup
           title: Yarn Setup
-      - yarn@0:
-          inputs:
-            - cache_local_deps: 'yes'
```
Comment (Cal-L) [inline bitrise.yml:53]: <3 removing code

**Review by Cal-L — APPROVED** [review]
If the comment regarding versions looks good, LGTM!

File: bitrise.yml (hunk 2 above)
Comment (andreahaku) [reply bitrise.yml:53]: yes finally :)

**Review by Cal-L — APPROVED** [review]
LGTM!

**Pull request merged**